src/main/scala/util/package.scala

```diff
@@ -234,23 +234,55 @@ package object util {
     map.view.map({ case (k, vs) => k -> vs.toList }).toList
   }
 
+/** provides operators useful for working with bidirectional [[Bundle]]s
+  * 
+  * In terms of [[Flipped]] with a producer 'p' and 'consumer' c:
+  * c :<= p // means drive all unflipped fields of 'c' from 'p' (e.g.: valid/bits)  
+  * c :=> p // means drive all flipped fields of 'c' from 'p' (e.g.: ready)  
+  * c :<> p // do both of the above
+  * p :<> c // do both of the above, but you'll probably get a Gender error later.
+  * 
+  * This utility class is needed because in [[chisel3]]:
+  * c := p // only works if there are no directions on fields.  
+  * c <> p // only works if one of those is an [[IO]] (not a [[Wire]]).
+  * 
+  * Compared with [[chisel3]] operators:
+  * c <> p   is an 'actual-direction'-inferred 'c :<> p' or 'p :<> c'
+  * c := p   is equivalent to 'c :<= p' + 'p :=> c'
+  * 
+  * Contrast this with 'c :<> p' which will connect a ready-valid producer
+  * 'p' to a consumer 'c'.
+  * If you flip this to 'p :<> c', it works the way you would expect (flipping the role of producer/consumer).
+  * This is how Chisel._ (compatability mode) and firrtl work.
+  * Some find that  ':<>' has superior readability (even if the direction can be inferred from an IO),
+  * because it clearly states the intended producer/consumer relationship. 
+  * You will get an appropriate error if you connected it the wrong way
+  * (usually because you got the IO direction wrong) instead of silently succeeding.
+  * 
+  * What if you want to connect all of the signals (e.g. ready/valid/bits)
+  * from producer 'p' to a monitor 'm'?
+  * For example in order to tap the connection to monitor traffic on an existing connection.
+  * In that case you can do 'm :<= p' and 'm :=> p'.
```

Comment:
```suggestion
  * In that case you can do 'm :<= p' and 'p :=> m'.
```

@terpstra can you double-check this one too? Just intuitively I feel like this is how the sentence should read... all the normal P signals go to M, and then also all other P signals go to M.

Reply:
based on my current understanding, the former is wrong. the latter might be correct (m.ready = p.ready) for a monitor.